### ISSUE TITLE
Way to use IndexController.refresh()?

### ISSUE THREAD

**Issue by cricket007** [YousefED/ElasticUI#64]
I am not too familiar with Angular, but I am looking for a way to periodically refresh the indexVM variable so the search results will refresh every few seconds. 

I have hooked a custom controller to the view with this code and I can see in the console that the `$scope.getData()` function gets called every 5 seconds. I would like to somehow call the `refresh` method on the `IndexController` here which I can see in the source is inside the `IndexDirective`. I've tried using `$scope.indexVM` and injecting the `$controller` service, but no success to get what I was looking for. 

### Controller

```
angular.module('demo.search', ['ngRoute', 'elasticui'])

    .config(['$routeProvider', function($routeProvider) {
        $routeProvider.when('/search', {
            templateUrl: 'search/search.html',
            controller: 'SearchCtrl'
        });
    }])

    .controller('SearchCtrl', ['$scope', '$timeout', function($scope, $timeout) {

        // Function to get the data
        $scope.getData = function(){
          console.log("Fetched data!");
        };

        // Function to replicate setInterval using $timeout service.
        $scope.intervalFunction = function(){
          $timeout(function() {
            $scope.getData();
            $scope.intervalFunction();
          }, 5000);
        };

        // Kick off the interval
        $scope.intervalFunction();
    }])
```

### HTML

I am using the SearchBoxDirective and the search results populate appropriately when I type in the input field. From what I can tell, the `eui-index` is the `IndexDirective` and somehow the `indexVM` variable down in the `ng-repeat` is being pulled from it. I just can't figure out the scoping of the Controllers and Directives (or if there is any link between them). 
```
<div class="container-fluid" eui-index="'demo'" eui-sort="ejs.Sort('datetime').order('desc').type('couchbaseDocument')" eui-enabled="true">
  <div class="row">
    <div class="col-md-12">
      <div class="panel panel-primary">
        <div class="panel-heading">Search</div>
        <div class="panel-body">
            <eui-searchbox field="'text'" doctype="'couchbaseDocument'"></eui-searchbox> 
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-md-12">
      <div class="panel panel-primary">
        <div class="panel-heading">Results</div>
          <div class="panel-body">
            <ul>
              <li ng-repeat="doc in indexVM.results.hits.hits">
                <p>{{doc._source.doc.text}}</p>
              </li>
            </ul>
            <eui-simple-paging></eui-simple-paging>
          </div>
      </div>
    </div>
  </div>
</div>
```

**Comment by YousefED** [comment]
I'm using the following code myself:

    $interval(function () {
                    if (_this.scope.session_settings.auto_refresh) {
                        _this.scope.indexVM.refresh();
                    }
                }, 1000 * 10);


Hope this helps!

**Comment by cricket007** [comment]
Looks promising, but where exactly do I place that? 

I tried putting just that `if` statement inside my `$scope.getData` then placing that entire block directly into my `SearchCtrl` and both times I got `ReferenceError: _this is not defined`.

**Comment by YousefED** [comment]
I think you should just be able to call $scope.indexVM.refresh() im your
getData function. Does that work or is indexVM not defined?

**Comment by cricket007** [comment]
Correct, `$scope.indexVM` is undefined.

**Comment by YousefED** [comment]
Hi Jordan,

I think this has to do with the Angular Scope chain.

In my app for example, I set up the main controller as follows:

<body ng-app="stats" ng-controller="MainController" eui-index="'indexname'">

then, I do have access to indexVM on the MainController scope.

I hope this helps + you can find a good way to access the indexVM object on your scope

**Comment by cricket007** [comment]
Okay cool! That seems to work. For future reference, since your HTML seems to be hidden in GitHub, here it is again: 

```html
<body ng-app="stats" ng-controller="MainController" eui-index="'indexname'">
```

I do have an `ng-view` within my `body` tag, though, and my Controller is hooked to that (the `ng-view` is what I posted before). I ended up moving the `eui-index` directive from the view into the `body` tag of my `index.html`.

Thanks!
